blockchain/chain_sync/src/sync.rs

```diff
@@ -98,16 +97,31 @@ where
         db: Arc<DB>,
         network_send: Sender<NetworkMessage>,
         network_rx: Receiver<NetworkEvent>,
+        genesis_buffer: Option<Cid>,
     ) -> Result<Self, Error> {
         let chain_store = ChainStore::new(db.clone());
-        let genesis = match chain_store.genesis()? {
-            Some(gen) => Tipset::new(vec![gen])?,
+
+        let genesis = match genesis_buffer {
+            Some(genesis_cid) => {
+                debug!("Initializing ChainSyncer with genesis from config");
+                let genesis_block =
+                    db.get(&genesis_cid)
+                    .map_err(|e| Error::Other(e.to_string()))?
+                    .ok_or_else(|| Error::Other("Could not find genesis block despite being loaded using a genesis file".to_owned()))?;
+                Tipset::new(vec![genesis_block])?
+            }
             None => {
-                // TODO change default logic for genesis or setup better initialization
-                warn!("No genesis found in data storage, using a default");
-                Tipset::new(vec![BlockHeader::default()])?
+                debug!("No specified genesis in config. Attempting to load from store");
+                match chain_store.genesis()? {
+                    Some(store_genesis) => Tipset::new(vec![store_genesis])?,
+                    None => {
+                        warn!("No genesis provided by config or blockstore, using default Tipset");
+                        Tipset::new(vec![BlockHeader::default()])?
+                    }
+                }
             }
         };
+        info!("Initializing ChainSyncer with genesis: {:?}", genesis);
```

Comment:
instead of debug printing the whole genesis tipset, can you just print out the genesis Cid? This floods the logs with unnecessary info

Reply:
Oops, my bad, yeah. I think it makes sense to leave it as an info print though...